planet/clients/orders.py

```diff
@@ -302,68 +302,51 @@ async def aggregated_order_stats(self) -> dict:
-            directory: Write to given directory instead of current directory.
+            order: a finished Order.
+            directory: a download directory Path.
             overwrite: Overwrite any existing files.
-            progress_bar: Show progress bar during download.
 
         Returns:
-            Path to downloaded file.
+            None.
 
         Raises:
-            planet.exceptions.APIException: On API error.
-        """
-        req = self._request(location, method='GET')
-
-        async with self._session.stream(req) as resp:
-            body = StreamingBody(resp)
-            dl_path = os.path.join(directory or '.', filename or body.name)
-            await body.write(dl_path,
-                             overwrite=overwrite,
-                             progress_bar=progress_bar)
-        return dl_path
+            OrderError
 
-    async def download_order(self,
-                             order_id: str,
-                             directory: str = None,
-                             overwrite: bool = False,
-                             progress_bar: bool = False) -> typing.List[str]:
-        """Download all assets in an order.
-
-        Parameters:
-            order_id: The ID of the order
-            directory: Write to given directory instead of current directory.
-            overwrite: Overwrite any existing files.
-            progress_bar: Show progress bar during download.
+        """
+        semaphore = asyncio.Semaphore(MAX_CONCURRENCY)
 
-        Returns:
-            Paths to downloaded files.
+        async with ClientSession() as session:
 
-        Raises:
-            planet.exceptions.APIException: On API error.
-        """
-        order = await self.get_order(order_id)
-        locations = order.locations
-        LOGGER.info(
-            f'downloading {len(locations)} assets from order {order_id}')
-        filenames = [
-            await self.download_asset(location,
-                                      directory=directory,
-                                      overwrite=overwrite,
-                                      progress_bar=progress_bar)
-            for location in locations
-        ]
-        return filenames
+            async def _save_asset(url: str, path: Path) -> None:
+                """Download resource at url and save to path directory."""
+                async with semaphore:
+                    filename = path.joinpath(Path(url).name)
+                    async with session.get(url) as resp:
+                        try:
+                            resp.raise_for_status()
+                        except Exception as service_err:
+                            raise OrderError(
+                                "Order asset fetch failed.") from service_err
+                        with open(filename, "wb") as f:
+                            async for data, _ in resp.content.iter_chunks():
+                                f.write(data)
```

Comment:
This function replaces download_asset(). The semaphore blocks if we've reached maximum concurrency. If our services responses are chunked, we can write the chunks as they come in.

Reply:
Overwriting flag not yet supported.